rbac/management/utils.py

```diff
@@ -114,9 +114,7 @@ def get_principal(
             )
         else:
             # Avoid possible race condition if the user was created while checking BOP
-            principal, _ = Principal.objects.get_or_create(
-                username=username, tenant=tenant, defaults={"user_id": request.user.user_id}
-            )  # pylint: disable=unused-variable
+            principal, _ = Principal.objects.get_or_create(username=username, tenant=tenant)
```

Comment:
@alechenninger this is all accounted for prior to getting here now with the changes in this PR, correct? Where this right now is a bug in that it could set the wrong `user_id` value. Is there still an edge-case where we'd create a duplicate `Principal` record for a principal where there wasn't a `user_id`? 

Actually, I'm thinking of the inverse, which I noted [here](https://github.com/RedHatInsights/insights-rbac/pull/1179#discussion_r1775068737) which should no longer be a problem based on this PR. Just thinking out loud. 😄

Reply:
It's a good question. We can create a record without a user_id (we'll populate it later if so). There shouldn't be a duplicate so long as username on the Principal is unique–which it is thankfully, so we _should_ be good there I think.